app/src/main/java/rick_and_morty/ui/episodes/EpisodesViewModel.kt

```diff
@@ -34,17 +34,13 @@ class EpisodesViewModel @Inject constructor(
         viewModelScope.launch {
             try {
                 var dbEpisodes = getEpisodesFromDatabase()
-                if (dbEpisodes.isEmpty()) {
-                    val apiEpisodes = episodesRepository.getEpisodes(page)
-                    saveEpisodesToDatabase(apiEpisodes)
-                    dbEpisodes = getEpisodesFromDatabase()
-                    page += 1
-                } else if (dbEpisodes.size < page * 20) {
+                if (dbEpisodes.isEmpty() || dbEpisodes.size < page * 20) {
```

Comment:
If the second option is true the first one will also be true. `dbEpisodes.isEmpty()` can be omitted.

Actually, do we need this check at all?

Reply:
done